pkg/scaling/resolver/scale_resolvers.go

```diff
@@ -193,8 +194,13 @@ func ResolveAuthRefAndPodIdentity(ctx context.Context, client client.Client, log
 					fmt.Errorf("error getting service account: '%s', error: %w", serviceAccountName, err)
 			}
 			authParams["awsRoleArn"] = serviceAccount.Annotations[kedav1alpha1.PodIdentityAnnotationEKS]
-		} else if podIdentity.Provider == kedav1alpha1.PodIdentityProviderAwsKiam {
+		case kedav1alpha1.PodIdentityProviderAwsKiam:
 			authParams["awsRoleArn"] = podTemplateSpec.ObjectMeta.Annotations[kedav1alpha1.PodIdentityAnnotationKiam]
+		case kedav1alpha1.PodIdentityProviderAzure:
+			if *podIdentity.IdentityID == "" {
```

Comment:
We have to check the pointer before dereferencing it:
```suggestion
			if podIdentity.IdentityID != nil && *podIdentity.IdentityID == "" {
```